File: csharp/DiagramPrinter/PhysicalPrinter.cs
namespace DiagramPrinter;

public enum Toner
{
    Black,
    Cyan,
    Magenta,
    Yellow
}

public class PhysicalPrinter
{
    public bool IsAvailable { get; protected set; }
    public int JobCount { get; }

    public Dictionary<Toner, int> TonerLevels { get; protected set; } = new()
    {
        { Toner.Black, 0},
        { Toner.Cyan, 0},
        { Toner.Magenta, 0},
        { Toner.Yellow, 0}
    };

    public virtual bool StartDocument(bool isSummary, bool isPdf, string name)
    {
        throw new NotImplementedException("Can't call this from a unit test");
    }

    public virtual void EndDocument()
    {
        throw new NotImplementedException("Can't call this from a unit test");
    }
}

File: java/src/main/java/sammancoaching/PhysicalPrinter.java
package sammancoaching;

public class PhysicalPrinter {
    protected boolean isAvailable;
    private final int jobCount;

    public PhysicalPrinter() {
        this.jobCount = 0; // Assuming job count is initialized to zero
    }

    public boolean isAvailable() {
        return isAvailable;
    }

    public int getJobCount() {
        return jobCount;
    }

    public boolean startDocument(boolean isSummary, boolean isPdf, String name) {
        throw new UnsupportedOperationException("Can't call this from a unit test");
    }

    public void endDocument() {
        throw new UnsupportedOperationException("Can't call this from a unit test");
    }
}

